apps/frontend/src/pages/project/project.tsx

```diff
@@ -240,11 +244,38 @@ const Project = (): JSX.Element => {
 
 	const hasProject = project !== null;
 
-	const hasManageAllProjectsPermission = checkHasPermission(
-		[PermissionKey.MANAGE_ALL_PROJECTS],
-		userPermissions,
+	const hasManagePermission = Boolean(
+		checkHasPermission(
+			[PermissionKey.MANAGE_ALL_PROJECTS, PermissionKey.MANAGE_PROJECT],
+			allPermissions,
+		) &&
+			hasProject &&
+			checkHasProjectPermission(authenticatedUser, project.id, [
+				PermissionKey.MANAGE_PROJECT,
+			]),
```

Comment:
Split all of this and bellow into different variables, it's hard to understand the code this way